### PR TITLE
Zeppelin 2701 - Set HTML title for notebook page

### PR DESCRIPTION
**Pull request by ebuildy**
### What is this PR for?

From https://issues.apache.org/jira/browse/ZEPPELIN-2701, this set a HTML title for notebook page, very good when you have multiple tabs opened.

### What type of PR is it?
Feature

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-2701

### How should this be tested?
- title is always blank except for notebook page

### CODE REVIEW

**Comment by soralee** [comment]
Hi, @ebuildy. Thanks for improving it. 
But Travis is fail, could you check this?

**Comment by ebuildy** [comment]
Ya I think it complains about https://travis-ci.org/ebuildy/zeppelin , some jobs have failed, I am checking it. thanks you.

**Comment by soralee** [comment]
You're welcome! 
Yeah, the current Travis is not good but it's working. 
If when your Travis become green or it is fail but it is not related your PR, I can test it out (:

**Comment by ebuildy** [comment]
I don't know why it fails here:

https://travis-ci.org/ebuildy/zeppelin/jobs/249550330#L776

``ERROR org.apache.zeppelin.spark.SparkVersion:63 - Can not recognize Spark version DEV-10.10. Assume it's a future release
java.lang.NumberFormatException: For input string: "DEV"``

Also here https://travis-ci.org/ebuildy/zeppelin/jobs/249550330#L1392

``java.lang.NullPointerException
	at scala.tools.nsc.interpreter.IMain$ReadEvalPrint.call(IMain.scala:784)``

I took the master branch, is it ok ?

**Comment by soralee** [comment]
Ah, you don't mind those lines because your 1.8 job became green light!
So, I think that it would be nice to run again about 1.3 and 1.5 jobs.
Could you try again, please?

**Comment by ebuildy** [comment]
Yeah all green now! (https://travis-ci.org/ebuildy/zeppelin/builds/249550322)

**Comment by soralee** [comment]
Oh, It's great! 
Let me test it out as soon as possible.
Thanks!

**Comment by soralee** [comment]
Could you remove the `{{ $root.pageTitle }}` word when loading page like this?
title

File: zeppelin-web/src/app/notebook/notebook.controller.js [hunk 1]
```diff
@@ -443,6 +443,9 @@ function NotebookCtrl ($scope, $route, $routeParams, $location, $rootScope,
     if ($scope.note.paragraphs && $scope.note.paragraphs[0]) {
       $scope.note.paragraphs[0].focus = true
     }
+
+    $rootScope.pageTitle = $scope.note.name
```
Comment (1ambda) [inline zeppelin-web/src/app/notebook/notebook.controller.js]: Hi, you might `angualr.$watch` to update only when `$scope.note.name` is valid.

File: zeppelin-web/src/app/notebook/notebook.controller.js [hunk 2]
```diff
@@ -62,6 +62,10 @@ function NotebookCtrl ($scope, $route, $routeParams, $location, $rootScope,
   $scope.currentRevision = 'Head'
   $scope.revisionView = isRevisionPath($location.path())
 
+  $scope.$watch('note', function (value) {
+    $rootScope.pageTitle = value ? value.name : ''
+  }, true)
+
```
Comment (soralee) [inline zeppelin-web/src/app/notebook/notebook.controller.js]: When I built on `zeppelin-web` in my local, a error is occurred like the following contents.
```
[INFO] /home/sora/zeppelin/zeppelin/zeppelin-web/src/app/notebook/notebook.controller.js
[INFO]    57:34  warning  Unnecessary escape character: \/  no-useless-escape
[INFO]    57:44  warning  Unnecessary escape character: \/  no-useless-escape
[INFO]    57:59  warning  Unnecessary escape character: \/  no-useless-escape
[INFO]    57:69  warning  Unnecessary escape character: \/  no-useless-escape
[INFO]    68:1   error    Trailing spaces not allowed       no-trailing-spaces
[INFO]   450:1   error    Trailing spaces not allowed       no-trailing-spaces
``` 
could you resolve this 68 line?
I guess the 450 line and your CI errors are related 68 line, so you can just remove space in 68 line.

**Comment by soralee** [comment]
I tested this one and it looks better than before, except above a my comment about CI failure.
And I suggest one thing. in this chance, how about changing landing page name from `localhost:8080/#/` to `Zeppelin` or something?
image

I know my suggestion is not related with this PR. But if you want to improve, you can make a PR or involve  code in this PR.

What do you think of?

**Comment by soralee** [comment]
Hey, @ebuildy!
If you don't want to improve my suggestion(#2458 (comment)), please let us know so that we will start merge process, I guess.

**Comment by ebuildy** [comment]
Nop sorry, this is very good suggestions, I was at Berlin for work, I am doing this now!

Thanks you,

**Comment by soralee** [comment]
Thanks a lots :slightly_smiling_face: 
Let me test it out again!

**Comment by 1ambda** [comment]
@ebuildy Thanks for the contribution.

Tested and works well, merge if no more discussion.

File: zeppelin-web/src/app/notebook/notebook.controller.js [hunk 3]
```diff
@@ -62,6 +62,10 @@ function NotebookCtrl ($scope, $route, $routeParams, $location, $rootScope,
   $scope.currentRevision = 'Head'
   $scope.revisionView = isRevisionPath($location.path())
 
+  $scope.$watch('note', function (value) {
+    $rootScope.pageTitle = value ? value.name : ''
```
Comment (1ambda) [inline zeppelin-web/src/app/notebook/notebook.controller.js]: Should be `Zeppelin` or something? when `value` is invalid.

**Comment by 1ambda** [comment]
@ebuildy I left a comment. Could u check that?

**Comment by soralee** [comment]
Tested and LGTM

**Comment by soralee** [comment]
@ebuildy could you check CI?

**Comment by 1ambda** [comment]
CI is green, I think jenkins was restarted somehow. 

- https://travis-ci.org/ebuildy/zeppelin/builds/253975556

LGTM merge if no more discussion.

**Comment by ebuildy** [comment]
I don't know why Jenkins is red, Travis is all green => https://travis-ci.org/ebuildy/zeppelin

**Pull request closed without merging**